tests/api/api.spec.ts

```diff
@@ -1145,6 +1145,51 @@ describe("@dataform/api", () => {
         expect(e.message).to.equal("Compilation timed out");
       }
     });
+
+    it("dataform.json is valid", async () => {
+      try {
+        utils.validate(
+          await compile({
+            projectDir: path.resolve("df/examples/dataform_json_checks/invalid_warehouse")
+          })
+        );
+        fail("Should have failed.");
+      } catch (e) {
+        expect(e.message).to.match(/Invalid value on property warehouse: dataform/);
+      }
+      try {
+        utils.validate(
+          await compile({
+            projectDir: path.resolve("df/examples/dataform_json_checks/missing_warehouse")
+          })
+        );
+        fail("Should have failed.");
+      } catch (e) {
+        expect(e.message).to.match(/Missing mandatory property: warehouse/);
+      }
+      try {
+        utils.validate(
+          await compile({
+            projectDir: path.resolve("df/examples/dataform_json_checks/invalid_defaultschema")
+          })
+        );
+        fail("Should have failed.");
+      } catch (e) {
+        expect(e.message).to.match(
+          /Invalid value on property defaultSchema: rock&roll. Should only contain alphanumeric characters, underscores and\/or hyphens./
+        );
+      }
+      try {
+        utils.validate(
+          await compile({
+            projectDir: path.resolve("df/examples/dataform_json_checks/valid_dataform_json")
+          })
+        );
+        fail("Forcing a failure.");
```

Comment:
just don't bother wrapping this case in a `try/fail/catch` at all; if `compile` throws (without a wrapping `try/catch`) then the test case will automatically fail.

Reply:
Done